internal/provider/validators.go

```diff
@@ -106,3 +108,39 @@ func validateDatacenter(v interface{}, path cty.Path) diag.Diagnostics {
 
 	return diagnostics
 }
+
+func validateConsulClusterTier(v interface{}, path cty.Path) diag.Diagnostics {
```

Comment:
Very cool pattern for these `validate*` functions, thanks for adding them (and tests)!